programs/bpf_loader/src/lib.rs

```diff
@@ -307,11 +321,13 @@ pub fn create_ebpf_vm<'a, 'b>(
     orig_account_lengths: Vec<usize>,
     invoke_context: &'a mut InvokeContext<'b>,
 ) -> Result<EbpfVm<'a, RequisiteVerifier, InvokeContext<'b>>, EbpfError> {
-    let _ = invoke_context.consume_checked(
-        ((heap.len() as u64).saturating_div(32_u64.saturating_mul(1024)))
-            .saturating_sub(1)
-            .saturating_mul(invoke_context.get_compute_budget().heap_cost),
-    );
+    let _ = invoke_context.consume_checked(calculate_heap_cost(
```

Comment:
Now that we are feature gating this anyway, we should also stop discarding / ignoring the result. In other words `let _ =` should become `let heap_cost_result = ...; if feature { heap_cost_result?; }`.

Reply:
Yea sure, added it.